### PR TITLE
Buffer ReverseProxy use http.ReverseProxy 

### PR DESCRIPTION
**Pull request by juliens**
This PR merge serveBufferHTTP and serveStreamHTTP to both use the http.ReverseProxy (as http.ReverseProxy without flushInterval works in "buffer" mode)
 We fix XForwarded too because some parts are already in http.ReverseProxy

http.ReverseProxy don't have `ErrorHandling`, that's why I have wrote a [PR on go](https://go-review.googlesource.com/c/go/+/77410).
In the meantime, I have proxied the RoundTripper in order to catch the errors.
With this, there is a little breaking change, in the ErrorHandler.serveHTTP, you can not Hijack/Flush anymore.

### CODE REVIEW

File: forward/fwd.go [hunk 1]
```diff
@@ -150,6 +174,12 @@ func New(setters ...optSetter) (*Forwarder, error) {
 		}
 	}
 
+	if !f.stream {
+		f.flushInterval = 0
+	} else if f.flushInterval == 0 {
+		f.flushInterval = time.Duration(100) * time.Millisecond
```
Comment (mmatur) [inline forward/fwd.go]: You should maybe create a constant to define this default flush interval value

File: forward/fwd.go [hunk 2]
```diff
@@ -183,98 +219,30 @@ func (f *Forwarder) ServeHTTP(w http.ResponseWriter, req *http.Request) {
...
-}
-
-// copyRequest makes a copy of the specified request to be sent using the configured
-// transport
-func (f *httpForwarder) copyRequest(req *http.Request, u *url.URL) *http.Request {
-	outReq := new(http.Request)
-	*outReq = *req // includes shallow copies of maps, but we handle this below
-
-	outReq.URL = utils.CopyURL(req.URL)
+func (f *httpForwarder) modifyRequest(outReq *http.Request, u *url.URL) {
```
Comment (mmatur) [inline forward/fwd.go]: Could you please add a comment

File: forward/rewrite.go [hunk 3]
```diff
@@ -46,9 +57,11 @@ func (rw *HeaderRewriter) Rewrite(req *http.Request) {
 		}
 	}
 
-	if xfh := req.Header.Get(XForwardedHost); xfh != "" && rw.TrustForwardHeader {
-		req.Header.Set(XForwardedHost, xfh)
-	} else if req.Host != "" {
+	if xfp := req.Header.Get(XForwardedPort); xfp == "" {
```
Comment (mmatur) [inline forward/rewrite.go]: Could you please rename `xfp` to `xfPort`

**Review by mmatur — CHANGES_REQUESTED** [review]

**Review by mmatur — APPROVED** [review]
LGTM :clap:

**Review by nmengin — APPROVED** [review]
LGTM

**Review by ldez — APPROVED** [review]
LGTM

**Pull request closed without merging**